string joints
string terminatingCondition
string resetPOS
string pCMD
string min_thresh
string bias
float32 tol
